### PR TITLE
Improve docs and add examples for Visitor

### PR DESCRIPTION
**Pull request by alamb**
Follow on to https://github.com/sqlparser-rs/sqlparser-rs/pull/765

### CODE REVIEW

**Comment by alamb** [comment]
cc @tustvold

**Comment by alamb** [comment]
@tustvold do you have time to review this PR? I hope to merge this later today / make a release

File: src/ast/visitor.rs [hunk 1]
```diff
@@ -57,8 +69,70 @@ visit_noop!(u8, u16, u32, u64, i8, i16, i32, i64, char, bool, String);
 #[cfg(feature = "bigdecimal")]
 visit_noop!(bigdecimal::BigDecimal);
 
-/// A visitor that can be used to walk an AST tree
+/// A visitor that can be used to walk an AST tree.
+///
+/// `previst_` methods are invoked before visiting all children of the
+/// node and `postgisit_` methods are invoked after visiting all
```
Comment (tustvold) [inline src/ast/visitor.rs:75]:
```suggestion
/// node and `postvisit_` methods are invoked after visiting all
```

File: src/ast/visitor.rs [hunk 2]
```diff
@@ -102,7 +176,33 @@ impl<E, F: FnMut(&ObjectName) -> ControlFlow<E>> Visitor for RelationVisitor<F>
...
+/// # use core::ops::ControlFlow;
+/// let sql = "SELECT a FROM foo where x IN (SELECT y FROM bar)";
+/// let statements = Parser::parse_sql(&GenericDialect{}, sql)
+///    .unwrap();
+///
+/// // visit statements, capturing relations (table names)
+/// let mut visited = vec![];
+/// visit_relations(&statements, |relation| {
+///   visited.push(format!("RELATION: {}", relation));
+///   ControlFlow::<()>::Continue(())
```
Comment (tustvold) [inline src/ast/visitor.rs:195]: I'm surprised you need to type hint this

File: src/ast/visitor.rs [hunk 3]
```diff
@@ -144,7 +272,33 @@ impl<E, F: FnMut(&Statement) -> ControlFlow<E>> Visitor for StatementVisitor<F>
     }
 }
 
-/// Invokes the provided closure on all statements present in v
+/// Invokes the provided closure on all statements (e.g. `SELECT`, `CREATE TABLE`, etc)present in `v`
```
Comment (tustvold) [inline src/ast/visitor.rs:275]:
```suggestion
/// Invokes the provided closure on all statements (e.g. `SELECT`, `CREATE TABLE`, etc) present in `v`
```

**Review by tustvold — APPROVED** [review]

File: src/ast/visitor.rs (hunk 2 above)
Comment (alamb) [reply src/ast/visitor.rs:195]: Yeah, me too -- I think it is because the return value is not used

**Pull request merged**